gateway/gateway-requests/src/registration/handshake/state.rs

```diff
@@ -36,6 +88,7 @@ pub(crate) struct State<'a, S> {
     /// The known or received public identity key of the remote.
     /// Ideally it would always be known before the handshake was initiated.
     remote_pubkey: Option<identity::PublicKey>,
+    coconut_credential: Option<Credential>,
```

Comment:
Thinking about it, I don't think coconut credential should be part of the state at all. It's only ever used once to get verified and is not required to be kept throughout the entire handshake (I guess another thing to change in the future, if you could leave a comment or something indicating that, it'd be great)

Reply:
Sure, I'll save all future work comments from this PR into github issues that we can (hopefully) take on in the sprints.